src/main/java/apaintus/controllers/CanvasController.java

```diff
@@ -53,45 +57,57 @@ public void initialize() {
         canvas.setOnMousePressed(event -> {
             lastMouseClickPosition = new Point(event.getX(), event.getY());
 
+            if (activeShape != null) {
+                activeShape.setSelected(false);
+                this.attributeController.resetAttributes();
+                redrawCanvas();
```

Comment:
i get the redraw is to remove the bounding box, but is this the good way to do so ? imagine if we have 189273409871239487 shapes, it will lag juste to unselect the shape and remove the Bbox

Reply:
T'as raison @pspino , mais le probleme est comment le canvas prend du data. Il fait juste empiler des nouvelles couches de drawLayer a chaque saveState(), donc pour enlever quelque chose dans le canvas de base, il faut le redessiner au complet sans cette chose. Possiblement, on pourrait avoir un systeme de layers qui forment un "canvas" ensemble et simplement redraw la layer concernee, mais c'est pas implemente